spec/automation/unit/method_validation/validate_quota_spec.rb

```diff
@@ -129,17 +138,77 @@ def run_automate_method
   end
 
   context "check_quota for VMReconfigure" do
+    let(:used) { YAML.dump(:storage => 10_240, :vms => 2, :cpu => 2, :memory => 1024) }
+    let(:requested) { YAML.dump(:storage => 20_480, :vms => 0, :cpu => 0, :memory => 4096) }
+
     it "check_quota false " do
-      ws = run_automate_method
+      setup_model("vmware_reconfigure")
+      ws = run_automate_method(reconfigure_attrs, used, requested)
       expect(ws.root['ae_result']).to eq('ok')
       expect(ws.root['check_quota']).to be_nil
     end
 
     it "check_quota true " do
+      setup_model("vmware_reconfigure")
       @extra_attrs = "check_quota=true"
-      ws = run_automate_method
+      ws = run_automate_method(reconfigure_attrs, used, requested)
       expect(ws.root['ae_result']).to eq('ok')
       expect(ws.root['check_quota']).to eq('true')
     end
+
+    it "failure max memory" do
+      setup_model("vmware_reconfigure")
+      @quota_limit_max  = YAML.dump(:storage => 0, :vms => 1, :cpu => 1, :memory => 4096)
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      err_msg = "Request exceeds maximum allowed for the following:" \
+                " (memory - Used: 1 KB plus requested: 4 KB exceeds quota: 4 KB) "
+      ws = run_automate_method(reconfigure_attrs, used, requested)
+      expect(ws.root['ae_result']).to eql('error')
+      @miq_request.reload
+      expect(@miq_request.options[:quota_max_exceeded]).to eql(err_msg)
+      expect(@miq_request.message).to eql(err_msg)
+    end
+
+    it "failure max memory" do
+      setup_model("vmware_reconfigure")
+      @quota_limit_max  = YAML.dump(:storage => 0, :vms => 1, :cpu => 1, :memory => 4096)
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      err_msg = "Request exceeds maximum allowed for the following:" \
+                " (memory - Used: 1 KB plus requested: 4 KB exceeds quota: 4 KB) "
+      ws = run_automate_method(reconfigure_attrs, used, requested)
+      expect(ws.root['ae_result']).to eql('error')
+      @miq_request.reload
+      expect(@miq_request.options[:quota_max_exceeded]).to eql(err_msg)
+      expect(@miq_request.message).to eql(err_msg)
+    end
+
+    it "failure max storage" do
+      setup_model("vmware_reconfigure")
+      @quota_limit_max  = YAML.dump(:storage => 20_480, :vms => 1, :cpu => 1, :memory => 0)
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      err_msg = "Request exceeds maximum allowed for the following:" \
+                " (storage - Used: 10 KB plus requested: 20 KB exceeds quota: 20 KB) "
+      ws = run_automate_method(reconfigure_attrs, used, requested)
+      expect(ws.root['ae_result']).to eql('error')
+      @miq_request.reload
+      expect(@miq_request.options[:quota_max_exceeded]).to eql(err_msg)
+      expect(@miq_request.message).to eql(err_msg)
+    end
```

Comment:
@billfitzgerald0120 can we make this a shared example and feed different values into it.

Reply:
@mkanoor Created 2 shared examples, one for Provisioning and one for Reconfigure